### DISCUSSION TITLE
Unit 4.3 Zero-rule baseline (majority classifier)

### DISCUSSION THREAD

**Discussion by harishb00** [Lightning-AI/dl-fundamentals#97] [Lecture videos and quizzes] [answered]
Below image is a snapshot from [MNIST MLP Notebook](https://github.com/Lightning-AI/dl-fundamentals/blob/main/unit04-multilayer-nets/4.3-mlp-pytorch/4.3-mlp-pytorch-part3-5-mnist/4.3-mlp-pytorch-part5-mnist.ipynb)
![image](https://github.com/Lightning-AI/dl-fundamentals/assets/43300971/cea298ab-05c8-4f19-aa2f-951c8fbfb3a0)

@rasbt 
**Shouldn't the `majority_class` be from training dataset? Shouldn't it be `majority_class = train_counter.most_common(1)[0]`?**
Because from the training set we look for the majority class and predict the same for the real world unseen data.

**Comment by rasbt** [comment, marked as answer]
Good point. In a practical setting where you want to implement a MajorityClass classifier, I'd say you are absolutely right. Here, I was more focused on seeing the best possible test accuracy without actually using a model. The majority class derived from the training set would always be equal or lower than the one derived from a test set, so it's more like the "best lucky guess of a majority class classifier" here.
